pytorch_lightning/trainer/predict_loop.py

```diff
@@ -24,6 +27,27 @@ def __init__(self, trainer):
         self.max_batches = None
         self.num_dataloaders = None
         self.warning_cache = WarningCache()
+        self.batch_indices: Optional[List[int]] = []
+        self._return_predictions = not trainer.training_type_plugin.use_spawn
+        self._previous_grad_status: Optional[bool] = None
+
+    @property
+    def return_predictions(self) -> bool:
+        return self._return_predictions
+
+    @return_predictions.setter
+    def return_predictions(self, return_predictions: Optional[bool]) -> None:
```

Comment:
why optional bool for `return_predictions`?

Reply:
By default, `return_predictions` is set to None in the `trainer.predict` as it depends on the selected accelerator.
Spawn doesn't support it.